docs/user-guide/kubeadm-reference.md

```diff
@@ -0,0 +1,95 @@
+# kubeadm reference
+
+This document specifies the advanced options of kubeadm.
+
+## Usage
+
+### `kubeadm init`
+
+It is usually sufficient to run `kubeadm init` with the default parameters,
+but in some cases you might like to override the default behaviour.
+Here we specify all the flags that can be used to customise the Kubernetes
+installation.
+
+- `--api-advertise-addresses` (multiple values are allowed by having multiple
+flag declarations or multiple values separated by commas)
+- `--api-external-dns-names` (multiple values are allowed by having multiple
+flag declarations or multiple values separated by commas)
+
+By default, `kubeadm` will automatically detect IP addresses and use
+these to generate certificates for the API server. If you would like
+to access the API server through an IP address that can't be automatically
+detected, or through a hostname, you can override these defaults with
+`--api-advertise-addresses` and `--api-external-dns-names`.
+
+- `--cloud-provider`
+
+Currently, `kubeadm` does not provide autodetection of cloud provider.
+This means that load balancing and persistent volumes are not supported out
+of the box. You can specify a cloud provider using `--cloud-provider`.
+Valid values are the ones supported by `controller-manager`, namely `"aws"`,
+`"azure"`, `"cloudstack"`, `"gce"`, `"mesos"`, `"openstack"`, `"ovirt"`,
+`"rackspace"`, `"vsphere"`.
+
+- `--external-etcd-cafile` etcd certificate authority certificate file
+- `--external-etcd-certfile` etcd client certificate file
+- `--external-etcd-endpoints` comma separated list of etcd endpoints to use
```

Comment:
To confirm, does the `Fields that support multiple values will do so either with comma
separation, or by specifying the flag multiple times.` rule apply to `--external-etcd-endpoints`? If not I think it should do for a consistent UX